lib/pure/asyncdispatch.nim

```diff
@@ -368,15 +368,21 @@ proc all*[A](futs: seq[Future[A]]): Future[seq[A]] =
     completedFutures = 0
 
   for i, fut in futs:
-    fut.callback = proc(f: Future[A]) =
-      retValues[i] = f.read()
-      inc(completedFutures)
+    proc setCallback(i: int) =
+      fut.callback = proc(f: Future[A]) =
+        retValues[i] = f.read()
+        inc(completedFutures)
 
-      if completedFutures == len(futs):
-        retFuture.complete(retValues)
+        if completedFutures == len(retValues):
+          retFuture.complete(retValues)
+
+    setCallback(i)
 
   return retFuture
 
+proc all*[A](futs: varargs[Future[A]]): Future[seq[A]] =
+  return all(@futs)
```

Comment:
@moigagoo, i guess you should now be fine with varargs, because you no longer capture it. In your first pr you were trying to capture it in the closure. As far as i understood. =)